### DISCUSSION TITLE
Would it be possible for Applications Blur to change apps opacity but not text opacity?

### DISCUSSION THREAD

**Discussion by gabrielgcma** [aunetx/blur-my-shell#504] [Q&A] [answered]
I'm a total newbie when it comes to display servers, GNOME Shell and whatnot. Applications Blur is working perfectly fine and it's beautiful.

On some specific applications with no native transparency option provided (VSCode, for example), I need to tinker around with the **Opacity** property in order to get transparency (and blur, of course).

VSCode with 255 Opacity:

![image](https://github.com/aunetx/blur-my-shell/assets/93090939/20a2d250-2e0e-4124-b89f-01d2f1c30204)

VSCode with 205 Opacity:

![image](https://github.com/aunetx/blur-my-shell/assets/93090939/bad9478e-c09a-4ffc-939a-676bcb12e8dd)

Which has a significant impact on text opacity.

I know it says right there that it affects legibility, but is that a matter of implementing a new feature or is it the way extensions are limited to act upon the window?

**Comment by aunetx** [comment, marked as answer]
Hello, thanks for opening this discussion!

Blur my Shell does not provide this option, as it is unable to change the application's style. The most it can do is to decrease the opacity of the whole application, but that includes the text of the app, and there is no way to do it differently.

However, it is still possible to do it by using a custom gtk theme, which would by itself decrease the background opacity of the application, and the enable application blur on BmS with a full opacity -- this is because the gtk theme has access to the application's style.

I don't have the name of any theme that does it, but I suppose there are a lot. If you find one that suits you (and optionnaly, that is similar to the base adwaita theme), I would be very glad to include a notice in blur-my-shell's application preferences, as you most likely are not the only person with this request!

**Reply by gabrielgcma** [reply]
Thanks for the reply! I'll have a look.
